### PR TITLE
Feature puma patches [1/x] refactoring

### PR DESCRIPTION
**Pull request by sandro-elsweijer**
**_Describe your changes here:_**
Doing some refactoring for the following PUMA patches PR

**_All these boxes must be checked by the reviewers before merging the pull request:_**

As a reviewer please read through all the code lines and make sure that the code is fully understood, bug free, well-documented and well-structured.

- [x] The author added a BSD statement to `doc/` (or already has one)
- [x] The code compiles without warning in debugging and release mode, with and without MPI (this should be executed automatically in a github action)

  If the Pull request introduces code that is not covered by the github action (for example coupling with a new library):
  - [x] Should this use case be added to the github action?
  - [x] If not, does the specific use case compile and all tests pass (check manually)

- [x] All tests pass (in various configurations, this should be executed automatically in a github action)
- [x] New source/header files are properly added to the Makefiles
- [x] New Datatypes are added to t8indent_custom_datatypes.txt
- [x] The reviewer executed the new code features at least once and checked the results manually
- [x] The code is covered in an existing or new test case
- [x] New tests use the Google Test framework
- [x] The code follows the [t8code coding guidelines](https://github.com/holke/t8code/wiki/Coding-Guideline)
- [x] The code is well documented
- [x] All function declarations, structs/classes and their members have a proper doxygen documentation
- [x] All new algorithms and data structures are sufficiently optimal in terms of memory and runtime (If this should be merged, but there is still potential for optimization, create a new issue)
- [x] If a new directory with source-files is added, it must be covered by the `script/find_all_source_files.scp` to check the indentation of these files.
- [x] If this PR introduces a new feature, it must be covered in an example/tutorial and a Wiki article.

### CODE REVIEW

File: src/t8_schemes/t8_default/t8_default_line/t8_dline_bits.h [hunk 1]
```diff
@@ -21,6 +21,7 @@
 */
 
 /** \file t8_dline_bits.h
+ * Definition if line-specific functions.
```
Comment (holke) [inline src/t8_schemes/t8_default/t8_default_line/t8_dline_bits.h:24]:
```suggestion
 * Definitions of line-specific functions.
```

File: src/t8_schemes/t8_default/t8_default_hex/t8_default_hex_cxx.cxx [hunk 2]
```diff
@@ -638,22 +638,23 @@ t8_default_scheme_hex_c::t8_element_vertex_coords (const t8_element_t *t,
...
+  t8_element_vertex_coords (elem, vertex, coords_int);
 
   /* We divide the integer coordinates by the root length of the hex
    * to obtain the reference coordinates. */
-  coords[0] = coords_int[0] / (double) P8EST_ROOT_LEN;
-  coords[1] = coords_int[1] / (double) P8EST_ROOT_LEN;
-  coords[2] = coords_int[2] / (double) P8EST_ROOT_LEN;
+  coords[0] /= (double) P8EST_ROOT_LEN;
+  coords[1] /= (double) P8EST_ROOT_LEN;
+  coords[2] /= (double) P8EST_ROOT_LEN;
```
Comment (holke) [inline src/t8_schemes/t8_default/t8_default_hex/t8_default_hex_cxx.cxx:657]: Careful.
This seems wrong.
`coords_int` is not `coords`

**Review by holke — CHANGES_REQUESTED** [review]
Great changes. Improves code readability a lot.

Some minor typos and two request where i think you shortened too much (the coords int in hex and quad).

File: src/t8_schemes/t8_default/t8_default_quad/t8_default_quad_cxx.cxx [hunk 3]
```diff
@@ -771,22 +771,23 @@ t8_default_scheme_quad_c::t8_element_vertex_coords (const t8_element_t *t,
...
   int                 coords_int[2];
-  t8_element_vertex_coords (t, vertex, coords_int);
+  t8_element_vertex_coords (elem, vertex, coords_int);
 
   /* We divide the integer coordinates by the root length of the quad
    * to obtain the reference coordinates. */
-  coords[0] = coords_int[0] / (double) P4EST_ROOT_LEN;
-  coords[1] = coords_int[1] / (double) P4EST_ROOT_LEN;
+  coords[0] /= (double) P4EST_ROOT_LEN;
+  coords[1] /= (double) P4EST_ROOT_LEN;
```
[the remark this answers is not in the document]
Comment (sandro-elsweijer) [reply src/t8_schemes/t8_default/t8_default_quad/t8_default_quad_cxx.cxx]: Oh yes, I totally missed that!

[the remark this answers is not in the document]
Comment (sandro-elsweijer) [reply src/t8_schemes/t8_default/t8_default_quad/t8_default_quad_cxx.cxx]: Fixed

File: src/t8_schemes/t8_default/t8_default_hex/t8_default_hex_cxx.cxx (hunk 2 above)
Comment (sandro-elsweijer) [reply src/t8_schemes/t8_default/t8_default_hex/t8_default_hex_cxx.cxx]: Fixed

**Review by holke — APPROVED** [review]

**Pull request merged**